pkg/vcdsdk/gateway.go

```diff
@@ -1949,3 +1950,40 @@ func (gm *GatewayManager) UpdateLoadBalancer(ctx context.Context, lbPoolName str
 	}
 	return vsSummary.VirtualIpAddress, nil
 }
+
+func (gm *GatewayManager) FetchIpSpacesBackingGateway(ctx context.Context) ([]string, error) {
+	ipSpaceService := gm.Client.APIClient.IpSpacesApi
+
+	filterString := fmt.Sprintf("gatewayId==%s", gm.GatewayRef.Id)
+	options := swaggerClient.IpSpacesApiGetFloatingIpSuggestionsOpts{Filter: optional.NewString(filterString), SortAsc: optional.NewString("ipSpaceRef.name"), SortDesc: optional.EmptyString()}
+	var pageSize int32 = 10
+	var pageNum int32 = 1
+	var resultTotal int32 = math.MaxInt32
+
+	var ipSpaceIds []string
+	for int32(math.Ceil(float64(resultTotal)/float64(pageSize))) >= pageNum {
+		suggestions, _, err := ipSpaceService.GetFloatingIpSuggestions(ctx, pageNum, pageSize, &options)
+		if err != nil {
+			return nil, err
```

Comment:
style nit: it will be helpful to add context to the error so that it is evident what the call was that failed (`fmt.Errorf("unable to get floating IP suggestion for gateway [%s]: [%v]", <gateway name>, err)`)

Else, we will get some 400 error deep in networking stack of VCD. Since we issue many network API calls one after the other, it will be tough to understand where the error arose, especially since we don't even have a debug stack in this case.